ui crashes when google serves a captcha page

- [yes] I carefully read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md) and agree to them.
--> I forgot to capture the crash report, error occurs randomly
- [yes] I checked if the issue/feature exists in the latest version. 16.2
- [ ] I did use the [incredible bugreport to markdown converter] (https://teamnewpipe.github.io/CrashReportToMarkdown/) to paste bug reports. (next time)

I'm using orbot. Once in a while newpipe crashes when loading a video. By following the share link to external browser I just found out that google asks to solve a captcha because of high site traffic (last tor exit node heavily in use?).
If it happens again I will post the crash report (probably useless) but also the served html (for the extractor to learn to recognize the code).

You were right, the UI error mentions the reCaptcha. After restarting Newpioe and opening the page in the browser I could save the html response. 
Maybe you can use it. The linked JavaScript file is short and should be downloadable. (Although it says in the first line PLEASE DON'T COPY AND PASTE THIS CODE :-) ). If not - let me know I saved the js file.

--------------- HTML Response (google.com - I'm in Germany) ---

<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html DIR="LTR">
<head><meta http-equiv="content-type" content="text/html; charset=utf-8"><meta name="viewport" content="initial-scale=1"><title>https://www.youtube.com/watch?v=bdEsVoqbD5s</title></head>
<body style="font-family: arial, sans-serif; background-color: #fff; color: #000; padding:20px; font-size:18px;" onload="e=document.getElementById('captcha');if(e){e.focus();}">
<div style="max-width:400px;">
<hr noshade size="1" style="color:#ccc; background-color:#ccc;"><br>
<form id="captcha-form" action="index" method="post">
<script src="https://www.google.com/recaptcha/api.js" async defer></script>
<script>var submitCallback = function(response) {document.getElementById('captcha-form').submit();};</script>
<div id="recaptcha" class="g-recaptcha" data-sitekey="6LfwuyUTAAAAAOAmoS0fdqijC2PbbdH4kjq62Y1b" data-callback="submitCallback" data-s="BBfKAu1Eom3pCRMZf7VDAbqXxfZhg_FZNClSE9ei8z7vM9o5g9Qgzq_ONxu1ADtQdJzBdu4LkFKQnUHYe_8J_R5DRc7qc1G2cpF8gjIyarRhzPTKU8d2v2xwkofySx3-0PddU7HOFFlR0JYyufh_kjRxr7Vs2BuQPuHdSHw0unQoVwAAobZpk8Ci-6B0NLtmwONU59kU1xBGCyq4l_KDSex3Ctt72SJo-pZvp3x369R6NEqrYUmEsLI"></div>
<input type='hidden' name='q' value='EgQzTTTYGNu73OoFIhkA8aeDS1a4MV1U262KuKjQoRIhE-o1nE_rMgFy'><input type="hidden" name="continue" value="https://www.youtube.com/watch?v=bdEsVoqbD5s">
</form>
<hr noshade size="1" style="color:#ccc; background-color:#ccc;">
<div style="font-size:13px;">
<b>Über diese Seite</b><br><br>Unsere Systeme haben ungewöhnlichen Datenverkehr aus Ihrem Computernetzwerk festgestellt. Diese Seite überprüft, ob die Anfragen wirklich von Ihnen und nicht von einem Robot gesendet werden. <a href="#" onclick="document.getElementById('infoDiv').style.display='block';">Warum?</a><br><br>
<div id="infoDiv" style="display:none; background-color:#eee; padding:10px; margin:0 0 15px 0; line-height:1.4em;">
Diese Seite wird angezeigt, wenn Google automatisch Anfragen aus Ihrem Computernetzwerk erkennt, die anscheinend gegen die <a href="//www.google.com/policies/terms/">Nutzungsbedingungen</a> verstoßen. Die Sperre läuft ab, sobald diese Anfragen eingestellt werden. In der Zwischenzeit können Sie unsere Dienste weiterhin mithilfe des oben stehenden CAPTCHAs nutzen.<br><br>Dieser Datenverkehr wurde möglicherweise von bösartiger Software, einem Browser-Plug-in oder einem Skript gesendet, das automatische Anfragen verschickt. Falls Sie Ihre Netzwerkverbindung mit anderen teilen, bitten Sie Ihren Administrator um Hilfe. Möglicherweise ist ein anderer Computer, der dieselbe IP-Adresse verwendet, für die Anfragen verantwortlich. <a href="//support.google.com/websearch/answer/86640">Weitere Informationen</a><br><br>Eventuell werden Sie um Eingabe des CAPTCHAs gebeten, weil Sie komplexe Anfragen verwenden, die bekanntermaßen von Robots verwendet werden, oder weil Sie sehr schnell Anfragen senden.

</div><br>
IP-Adresse: 51.77.52.216<br>Uhrzeit: 2019-08-16T21:19:29Z<br>URL: https://www.youtube.com/watch?v=bdEsVoqbD5s<br>
</div></div>
</body>
</html>

---- Error UI
## Exception
* __User Action:__ ui error
* __Request:__ App crash, UI failure
* __Content Language:__ GB
* __Service:__ none
* __Version:__ 0.17.0
* __OS:__ Linux Android 6.0.1 - 23


<details><summary><b>Crash log</b></summary><p>

```
java.lang.RuntimeException: Unable to start activity ComponentInfo{org.schabi.newpipe/org.schabi.newpipe.ReCaptchaActivity}: android.view.InflateException: Binary XML file line #14: Binary XML file line #9: Error inflating class EditText at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2416) at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2476) at android.app.ActivityThread.access$900(ActivityThread.java:150) at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1344) at android.os.Handler.dispatchMessage(Handler.java:102) at android.os.Looper.loop(Looper.java:148) at android.app.ActivityThread.main(ActivityThread.java:5417) at java.lang.reflect.Method.invoke(Native Method) at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726) at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616) at de.robv.android.xposed.XposedBridge.main(XposedBridge.java:107) Caused by: android.view.InflateException: Binary XML file line #14: Binary XML file line #9: Error inflating class EditText at android.view.LayoutInflater.inflate(LayoutInflater.java:539) at de.robv.android.xposed.XposedBridge.invokeOriginalMethodNative(Native Method) at de.robv.android.xposed.XposedBridge.handleHookedMethod(XposedBridge.java:360) at android.view.LayoutInflater.inflate(<Xposed>) at android.view.LayoutInflater.inflate(LayoutInflater.java:423) at android.view.LayoutInflater.inflate(LayoutInflater.java:374) at android.support.v7.app.AppCompatDelegateImpl.setContentView(AppCompatDelegateImpl.java:469) at android.support.v7.app.AppCompatActivity.setContentView(AppCompatActivity.java:140) at org.schabi.newpipe.ReCaptchaActivity.onCreate(ReCaptchaActivity.java:47) at android.app.Activity.performCreate(Activity.java:6292) at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1108) at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2369) ... 10 more Caused by: android.view.InflateException: Binary XML file line #9: Error inflating class EditText at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:782) at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:704) at android.view.LayoutInflater.rInflate(LayoutInflater.java:835) at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:798) at android.view.LayoutInflater.parseInclude(LayoutInflater.java:971) at de.robv.android.xposed.XposedBridge.invokeOriginalMethodNative(Native Method) at de.robv.android.xposed.XposedBridge.handleHookedMethod(XposedBridge.java:360) at android.view.LayoutInflater.parseInclude(<Xposed>) at android.view.LayoutInflater.rInflate(LayoutInflater.java:831) at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:798) at android.view.LayoutInflater.rInflate(LayoutInflater.java:838) at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:798) at android.view.LayoutInflater.parseInclude(LayoutInflater.java:971) at de.robv.android.xposed.XposedBridge.invokeOriginalMethodNative(Native Method) at de.robv.android.xposed.XposedBridge.handleHookedMethod(XposedBridge.java:360) at android.view.LayoutInflater.parseInclude(<Xposed>) at android.view.LayoutInflater.rInflate(LayoutInflater.java:831) at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:798) at android.view.LayoutInflater.inflate(LayoutInflater.java:515) ... 21 more Caused by: java.lang.UnsupportedOperationException: Failed to resolve attribute at index 50: TypedValue{t=0x2/d=0x7f040203 a=-1} at android.content.res.TypedArray.getDrawable(TypedArray.java:867) at android.content.res.XResources$XTypedArray.getDrawable(XResources.java:1363) at android.widget.TextView.<init>(TextView.java:884) at android.widget.EditText.<init>(EditText.java:64) at android.widget.EditText.<init>(EditText.java:60) at android.support.v7.widget.AppCompatEditText.<init>(AppCompatEditText.java:69) at android.support.v7.widget.AppCompatEditText.<init>(AppCompatEditText.java:65) at android.support.v7.app.AppCompatViewInflater.createEditText(AppCompatViewInflater.java:192) at android.support.v7.app.AppCompatViewInflater.createView(AppCompatViewInflater.java:114) at android.support.v7.app.AppCompatDelegateImpl.createView(AppCompatDelegateImpl.java:1266) at android.support.v7.app.AppCompatDelegateImpl.onCreateView(AppCompatDelegateImpl.java:1316) at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:746) ... 39 more android.view.InflateException: Binary XML file line #14: Binary XML file line #9: Error inflating class EditText at android.view.LayoutInflater.inflate(LayoutInflater.java:539) at de.robv.android.xposed.XposedBridge.invokeOriginalMethodNative(Native Method) at de.robv.android.xposed.XposedBridge.handleHookedMethod(XposedBridge.java:360) at android.view.LayoutInflater.inflate(<Xposed>) at android.view.LayoutInflater.inflate(LayoutInflater.java:423) at android.view.LayoutInflater.inflate(LayoutInflater.java:374) at android.support.v7.app.AppCompatDelegateImpl.setContentView(AppCompatDelegateImpl.java:469) at android.support.v7.app.AppCompatActivity.setContentView(AppCompatActivity.java:140) at org.schabi.newpipe.ReCaptchaActivity.onCreate(ReCaptchaActivity.java:47) at android.app.Activity.performCreate(Activity.java:6292) at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1108) at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2369) at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2476) at android.app.ActivityThread.access$900(ActivityThread.java:150) at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1344) at android.os.Handler.dispatchMessage(Handler.java:102) at android.os.Looper.loop(Looper.java:148) at android.app.ActivityThread.main(ActivityThread.java:5417) at java.lang.reflect.Method.invoke(Native Method) at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726) at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616) at de.robv.android.xposed.XposedBridge.main(XposedBridge.java:107) Caused by: android.view.InflateException: Binary XML file line #9: Error inflating class EditText at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:782) at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:704) at android.view.LayoutInflater.rInflate(LayoutInflater.java:835) at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:798) at android.view.LayoutInflater.parseInclude(LayoutInflater.java:971) at de.robv.android.xposed.XposedBridge.invokeOriginalMethodNative(Native Method) at de.robv.android.xposed.XposedBridge.handleHookedMethod(XposedBridge.java:360) at android.view.LayoutInflater.parseInclude(<Xposed>) at android.view.LayoutInflater.rInflate(LayoutInflater.java:831) at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:798) at android.view.LayoutInflater.rInflate(LayoutInflater.java:838) at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:798) at android.view.LayoutInflater.parseInclude(LayoutInflater.java:971) at de.robv.android.xposed.XposedBridge.invokeOriginalMethodNative(Native Method) at de.robv.android.xposed.XposedBridge.handleHookedMethod(XposedBridge.java:360) at android.view.LayoutInflater.parseInclude(<Xposed>) at android.view.LayoutInflater.rInflate(LayoutInflater.java:831) at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:798) at android.view.LayoutInflater.inflate(LayoutInflater.java:515) ... 21 more Caused by: java.lang.UnsupportedOperationException: Failed to resolve attribute at index 50: TypedValue{t=0x2/d=0x7f040203 a=-1} at android.content.res.TypedArray.getDrawable(TypedArray.java:867) at android.content.res.XResources$XTypedArray.getDrawable(XResources.java:1363) at android.widget.TextView.<init>(TextView.java:884) at android.widget.EditText.<init>(EditText.java:64) at android.widget.EditText.<init>(EditText.java:60) at android.support.v7.widget.AppCompatEditText.<init>(AppCompatEditText.java:69) at android.support.v7.widget.AppCompatEditText.<init>(AppCompatEditText.java:65) at android.support.v7.app.AppCompatViewInflater.createEditText(AppCompatViewInflater.java:192) at android.support.v7.app.AppCompatViewInflater.createView(AppCompatViewInflater.java:114) at android.support.v7.app.AppCompatDelegateImpl.createView(AppCompatDelegateImpl.java:1266) at android.support.v7.app.AppCompatDelegateImpl.onCreateView(AppCompatDelegateImpl.java:1316) at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:746) ... 39 more android.view.InflateException: Binary XML file line #9: Error inflating class EditText at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:782) at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:704) at android.view.LayoutInflater.rInflate(LayoutInflater.java:835) at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:798) at android.view.LayoutInflater.parseInclude(LayoutInflater.java:971) at de.robv.android.xposed.XposedBridge.invokeOriginalMethodNative(Native Method) at de.robv.android.xposed.XposedBridge.handleHookedMethod(XposedBridge.java:360) at android.view.LayoutInflater.parseInclude(<Xposed>) at android.view.LayoutInflater.rInflate(LayoutInflater.java:831) at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:798) at android.view.LayoutInflater.rInflate(LayoutInflater.java:838) at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:798) at android.view.LayoutInflater.parseInclude(LayoutInflater.java:971) at de.robv.android.xposed.XposedBridge.invokeOriginalMethodNative(Native Method) at de.robv.android.xposed.XposedBridge.handleHookedMethod(XposedBridge.java:360) at android.view.LayoutInflater.parseInclude(<Xposed>) at android.view.LayoutInflater.rInflate(LayoutInflater.java:831) at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:798) at android.view.LayoutInflater.inflate(LayoutInflater.java:515) at de.robv.android.xposed.XposedBridge.invokeOriginalMethodNative(Native Method) at de.robv.android.xposed.XposedBridge.handleHookedMethod(XposedBridge.java:360) at android.view.LayoutInflater.inflate(<Xposed>) at android.view.LayoutInflater.inflate(LayoutInflater.java:423) at android.view.LayoutInflater.inflate(LayoutInflater.java:374) at android.support.v7.app.AppCompatDelegateImpl.setContentView(AppCompatDelegateImpl.java:469) at android.support.v7.app.AppCompatActivity.setContentView(AppCompatActivity.java:140) at org.schabi.newpipe.ReCaptchaActivity.onCreate(ReCaptchaActivity.java:47) at android.app.Activity.performCreate(Activity.java:6292) at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1108) at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2369) at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2476) at android.app.ActivityThread.access$900(ActivityThread.java:150) at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1344) at android.os.Handler.dispatchMessage(Handler.java:102) at android.os.Looper.loop(Looper.java:148) at android.app.ActivityThread.main(ActivityThread.java:5417) at java.lang.reflect.Method.invoke(Native Method) at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726) at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616) at de.robv.android.xposed.XposedBridge.main(XposedBridge.java:107) Caused by: java.lang.UnsupportedOperationException: Failed to resolve attribute at index 50: TypedValue{t=0x2/d=0x7f040203 a=-1} at android.content.res.TypedArray.getDrawable(TypedArray.java:867) at android.content.res.XResources$XTypedArray.getDrawable(XResources.java:1363) at android.widget.TextView.<init>(TextView.java:884) at android.widget.EditText.<init>(EditText.java:64) at android.widget.EditText.<init>(EditText.java:60) at android.support.v7.widget.AppCompatEditText.<init>(AppCompatEditText.java:69) at android.support.v7.widget.AppCompatEditText.<init>(AppCompatEditText.java:65) at android.support.v7.app.AppCompatViewInflater.createEditText(AppCompatViewInflater.java:192) at android.support.v7.app.AppCompatViewInflater.createView(AppCompatViewInflater.java:114) at android.support.v7.app.AppCompatDelegateImpl.createView(AppCompatDelegateImpl.java:1266) at android.support.v7.app.AppCompatDelegateImpl.onCreateView(AppCompatDelegateImpl.java:1316) at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:746) ... 39 more java.lang.UnsupportedOperationException: Failed to resolve attribute at index 50: TypedValue{t=0x2/d=0x7f040203 a=-1} at android.content.res.TypedArray.getDrawable(TypedArray.java:867) at android.content.res.XResources$XTypedArray.getDrawable(XResources.java:1363) at android.widget.TextView.<init>(TextView.java:884) at android.widget.EditText.<init>(EditText.java:64) at android.widget.EditText.<init>(EditText.java:60) at android.support.v7.widget.AppCompatEditText.<init>(AppCompatEditText.java:69) at android.support.v7.widget.AppCompatEditText.<init>(AppCompatEditText.java:65) at android.support.v7.app.AppCompatViewInflater.createEditText(AppCompatViewInflater.java:192) at android.support.v7.app.AppCompatViewInflater.createView(AppCompatViewInflater.java:114) at android.support.v7.app.AppCompatDelegateImpl.createView(AppCompatDelegateImpl.java:1266) at android.support.v7.app.AppCompatDelegateImpl.onCreateView(AppCompatDelegateImpl.java:1316) at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:746) at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:704) at android.view.LayoutInflater.rInflate(LayoutInflater.java:835) at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:798) at android.view.LayoutInflater.parseInclude(LayoutInflater.java:971) at de.robv.android.xposed.XposedBridge.invokeOriginalMethodNative(Native Method) at de.robv.android.xposed.XposedBridge.handleHookedMethod(XposedBridge.java:360) at android.view.LayoutInflater.parseInclude(<Xposed>) at android.view.LayoutInflater.rInflate(LayoutInflater.java:831) at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:798) at android.view.LayoutInflater.rInflate(LayoutInflater.java:838) at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:798) at android.view.LayoutInflater.parseInclude(LayoutInflater.java:971) at de.robv.android.xposed.XposedBridge.invokeOriginalMethodNative(Native Method) at de.robv.android.xposed.XposedBridge.handleHookedMethod(XposedBridge.java:360) at android.view.LayoutInflater.parseInclude(<Xposed>) at android.view.LayoutInflater.rInflate(LayoutInflater.java:831) at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:798) at android.view.LayoutInflater.inflate(LayoutInflater.java:515) at de.robv.android.xposed.XposedBridge.invokeOriginalMethodNative(Native Method) at de.robv.android.xposed.XposedBridge.handleHookedMethod(XposedBridge.java:360) at android.view.LayoutInflater.inflate(<Xposed>) at android.view.LayoutInflater.inflate(LayoutInflater.java:423) at android.view.LayoutInflater.inflate(LayoutInflater.java:374) at android.support.v7.app.AppCompatDelegateImpl.setContentView(AppCompatDelegateImpl.java:469) at android.support.v7.app.AppCompatActivity.setContentView(AppCompatActivity.java:140) at org.schabi.newpipe.ReCaptchaActivity.onCreate(ReCaptchaActivity.java:47) at android.app.Activity.performCreate(Activity.java:6292) at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1108) at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2369) at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2476) at android.app.ActivityThread.access$900(ActivityThread.java:150) at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1344) at android.os.Handler.dispatchMessage(Handler.java:102) at android.os.Looper.loop(Looper.java:148) at android.app.ActivityThread.main(ActivityThread.java:5417) at java.lang.reflect.Method.invoke(Native Method) at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726) at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616) at de.robv.android.xposed.XposedBridge.main(XposedBridge.java:107) 
```
</p></details>
<hr>

--- a/app/build.gradle
+++ b/app/build.gradle
@@ -57,7 +57,7 @@ dependencies {
         exclude module: 'support-annotations'
     })
 
-    implementation 'com.github.TeamNewPipe:NewPipeExtractor:5f65788a2f89e'
+    implementation 'com.github.Stypox:NewPipeExtractor:06689a2f27edfe83bd4605a1cceed86c06a5ebf8'
 
     testImplementation 'junit:junit:4.12'
     testImplementation 'org.mockito:mockito-core:2.23.0'

--- a/app/src/main/java/org/schabi/newpipe/Downloader.java
+++ b/app/src/main/java/org/schabi/newpipe/Downloader.java
@@ -164,7 +164,7 @@ public class Downloader implements org.schabi.newpipe.extractor.Downloader {
         final ResponseBody body = response.body();
 
         if (response.code() == 429) {
-            throw new ReCaptchaException("reCaptcha Challenge requested");
+            throw new ReCaptchaException("reCaptcha Challenge requested", siteUrl);
         }
 
         if (body == null) {
@@ -214,7 +214,7 @@ public class Downloader implements org.schabi.newpipe.extractor.Downloader {
         final ResponseBody body = response.body();
 
         if (response.code() == 429) {
-            throw new ReCaptchaException("reCaptcha Challenge requested");
+            throw new ReCaptchaException("reCaptcha Challenge requested", siteUrl);
         }
 
         if (body == null) {
@@ -268,7 +268,7 @@ public class Downloader implements org.schabi.newpipe.extractor.Downloader {
         final ResponseBody body = response.body();
 
         if (response.code() == 429) {
-            throw new ReCaptchaException("reCaptcha Challenge requested");
+            throw new ReCaptchaException("reCaptcha Challenge requested", siteUrl);
         }
 
         if (body == null) {

--- a/app/src/main/java/org/schabi/newpipe/ReCaptchaActivity.java
+++ b/app/src/main/java/org/schabi/newpipe/ReCaptchaActivity.java
@@ -37,14 +37,23 @@ import android.webkit.WebViewClient;
  */
 public class ReCaptchaActivity extends AppCompatActivity {
     public static final int RECAPTCHA_REQUEST = 10;
+    public static final String RECAPTCHA_URL_EXTRA = "recaptcha_url_extra";
 
     public static final String TAG = ReCaptchaActivity.class.toString();
     public static final String YT_URL = "https://www.youtube.com";
+
+    private String url;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_recaptcha);
+
+        url = getIntent().getStringExtra(RECAPTCHA_URL_EXTRA);
+        if (url == null || url.isEmpty()) {
+            url = YT_URL;
+        }
+
 
         // Set return to Cancel by default
         setResult(RESULT_CANCELED);
@@ -73,15 +82,12 @@ public class ReCaptchaActivity extends AppCompatActivity {
         myWebView.clearHistory();
         android.webkit.CookieManager cookieManager = CookieManager.getInstance();
         if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
-            cookieManager.removeAllCookies(new ValueCallback<Boolean>() {
-                @Override
-                public void onReceiveValue(Boolean aBoolean) {}
-            });
+            cookieManager.removeAllCookies(aBoolean -> {});
         } else {
             cookieManager.removeAllCookie();
         }
 
-        myWebView.loadUrl(YT_URL);
+        myWebView.loadUrl(url);
     }
 
     private class ReCaptchaWebViewClient extends WebViewClient {

--- a/app/src/main/java/org/schabi/newpipe/fragments/BaseStateFragment.java
+++ b/app/src/main/java/org/schabi/newpipe/fragments/BaseStateFragment.java
@@ -180,7 +180,7 @@ public abstract class BaseStateFragment<I> extends BaseFragment implements ViewC
         }
 
         if (exception instanceof ReCaptchaException) {
-            onReCaptchaException();
+            onReCaptchaException((ReCaptchaException) exception);
             return true;
         } else if (exception instanceof IOException) {
             showError(getString(R.string.network_error), true);
@@ -190,11 +190,13 @@ public abstract class BaseStateFragment<I> extends BaseFragment implements ViewC
         return false;
     }
 
-    public void onReCaptchaException() {
+    public void onReCaptchaException(ReCaptchaException exception) {
         if (DEBUG) Log.d(TAG, "onReCaptchaException() called");
         Toast.makeText(activity, R.string.recaptcha_request_toast, Toast.LENGTH_LONG).show();
         // Starting ReCaptcha Challenge Activity
-        startActivityForResult(new Intent(activity, ReCaptchaActivity.class), ReCaptchaActivity.RECAPTCHA_REQUEST);
+        Intent intent = new Intent(activity, ReCaptchaActivity.class);
+        intent.putExtra(ReCaptchaActivity.RECAPTCHA_URL_EXTRA, exception.getUrl());
+        startActivityForResult(intent, ReCaptchaActivity.RECAPTCHA_REQUEST);
 
         showError(getString(R.string.recaptcha_request_toast), false);
     }

--- a/app/src/main/res/layout/activity_recaptcha.xml
+++ b/app/src/main/res/layout/activity_recaptcha.xml
@@ -1,16 +1,27 @@
 <?xml version="1.0" encoding="utf-8"?>
-<RelativeLayout
-    xmlns:android="http://schemas.android.com/apk/res/android"
+<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
     android:layout_width="match_parent"
     android:layout_height="match_parent"
     android:orientation="vertical">
+
+    <android.support.v7.widget.Toolbar
+        android:id="@+id/toolbar"
+        android:layout_width="match_parent"
+        android:layout_height="?attr/actionBarSize"
+        android:background="?attr/colorPrimary"
+        android:gravity="center_vertical"
+        android:minHeight="?attr/actionBarSize"
+        android:theme="@style/ThemeOverlay.AppCompat.ActionBar"
+        app:popupTheme="@style/ThemeOverlay.AppCompat.ActionBar"
+        app:titleTextAppearance="@style/Toolbar.Title">
+
+    </android.support.v7.widget.Toolbar>
 
     <WebView
         android:id="@+id/reCaptchaWebView"
         android:layout_width="match_parent"
         android:layout_height="match_parent"
         android:layout_marginTop="?attr/actionBarSize"/>
-
-    <include layout="@layout/toolbar_layout"/>
 
 </RelativeLayout>

